include/common/core/spi.hpp

```diff
@@ -26,8 +26,8 @@ class TMC2130Spi {
      * @param transmit The transmit buffer.
      * @param receive The receive buffer.
      */
-    virtual void transmit_receive(const TMC2130Spi::BufferType& transmit,
-                                  TMC2130Spi::BufferType& receive) = 0;
+    virtual auto transmit_receive(const TMC2130Spi::BufferType& transmit,
+                                  TMC2130Spi::BufferType& receive) -> bool;
```

Comment:
should this still be a pure virtual function?virtual auto transmit_receive(const ```
TMC2130Spi::BufferType& transmit,
                                  TMC2130Spi::BufferType& receive) -> bool = 0;
```

Reply:
Since both the spi and i2c devices only get built once, I think it's fine to keep this virtual.

Reply:
I don't think I see an implementation of `TMC2130Spi::transmit_receive`, just in the subclasses. In that, case we definitely need to keep it pure virtual and add an `= 0`.